tree-sitter-stack-graphs/src/loader.rs

```diff
@@ -105,13 +105,17 @@ impl FileAnalyzers {
         }
     }
 
-    pub fn add(mut self, file_name: String, analyzer: impl FileAnalyzer + 'static) -> Self {
+    pub fn add(
+        mut self,
+        file_name: String,
+        analyzer: impl FileAnalyzer + Send + Sync + 'static,
+    ) -> Self {
         self.file_analyzers.insert(file_name, Arc::new(analyzer));
         self
     }
 
-    pub fn get(&self, file_name: &str) -> Option<&Arc<dyn FileAnalyzer>> {
-        self.file_analyzers.get(file_name)
+    pub fn get(&self, file_name: &str) -> Option<Arc<dyn FileAnalyzer + Send + Sync>> {
+        self.file_analyzers.get(file_name).cloned()
```

Comment:
Any possibility for performance issues here? I don’t know off the top of my head how hot of a path this is.

Reply:
Good question! I don't think that is going to be a problem. This would be called once per file, and only for files that actually match the filename, i.e., that are going to use the result.